backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java

```diff
@@ -155,8 +177,8 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
                 .descriptionId(nodeDescription.getId())
                 .borderNode(containmentKind == NodeContainmentKind.BORDER_NODE)
                 .style(style)
-                .position(Position.UNDEFINED)
-                .size(Size.UNDEFINED)
+                .position(optionalPreviousNode.map(Node::getPosition).orElseGet(() -> nodePositionProvider.getNextPosition()))
+                .size(optionalPreviousNode.map(Node::getSize).orElseGet(() -> nodeSizeProvider.getSize(nodeChildren)))
```

Comment:
Extract that in variables

Reply:
fixed in 0d825fc825323b2902b76c564e851eec89a13e82